### PR TITLE
Fix #2

### LINKED ISSUE #2: Shift FileGopherServer in a class file
The current `FileGopherServer` class is in `server.cpp`

It would be better if it is in separate class file.

Need to create following files with proper code:
+ `FileGopherServer.h`
+ `FileGopherServer.cpp`

And include it in server.cpp file

### PR DESCRIPTION
**Pull request by skprabhanjan**
@Hrily , This PR fixes #2 .
The new command to compile will be  **g++ -o server server.cpp FileGopherServer.cpp -std=gnu++11**
Please review this and let me know if anything needs to changed.
Thanks :)

### CODE REVIEW

File: FileGopherServer.cpp [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+ * @param host The host where the listing resides.
+ * @param port The port where the listing resides.
+ */
+string FileGopherServer::create_line(char type, string user_string,
+                                     string path, string host, int port)
+{
+    // Convert integer port to string.
+    char ch_port[6];
+    sprintf(ch_port, "%d", port);
+    return string(1, type) + user_string + TAB + path + TAB + host + TAB + string(ch_port) + CRLF;
```
Comment (Hrily) [inline FileGopherServer.cpp]: Cut this line so that it comes within 80 char/line limit

File: FileGopherServer.cpp [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    struct hostent *hostentry;
+    hostentry = gethostbyname(hostname);
+    char *ipbuf;
+    ipbuf = inet_ntoa(*((struct in_addr *)hostentry->h_addr_list[0]));
+    printf("%s\n", ipbuf);
+    this->socket_host = string(ipbuf);
+    // Make the server gopher+ complaint.
+    string gopher_plus_content = "";
+    gopher_plus_content += "+-1" + CRLF;
+    gopher_plus_content += "+INFO: 1Main menu (non-gopher+)		0.0.0.0	7070" + CRLF;
```
Comment (Hrily) [inline FileGopherServer.cpp]: Same comment as above

**Comment by Hrily** [comment]
> The new command to compile will be g++ -o server server.cpp FileGopherServer.cpp -std=gnu++11

Can you update the README too to reflect this change?

File: FileGopherServer.cpp (hunk 1 above)
Comment (skprabhanjan) [reply FileGopherServer.cpp]: Done

File: FileGopherServer.cpp (hunk 2 above)
Comment (skprabhanjan) [reply FileGopherServer.cpp]: Done

**Comment by skprabhanjan** [comment]
@Hrily , All changes are done :)

File: FileGopherServer.cpp (hunk 1 above)
Comment (Hrily) [reply FileGopherServer.cpp]: Is it?

I can't see the change

**Comment by Hrily** [comment]
Thank you for your contribution :+1:

**Pull request merged**